lib/block-supports/layout.php

```diff
@@ -246,12 +246,16 @@ function gutenberg_restore_image_outer_container( $block_content, $block ) {
 		return $block_content;
 	}
 
+	$block_classnames                 = isset( $block['attrs']['className'] ) ? $block['attrs']['className'] : '';
+	$remove_classnames_pattern        = '/(<figure.*?class=".*)' . $block_classnames . '(.*?">)/';
+	$content_without_block_classnames = preg_replace( $remove_classnames_pattern, '$1 $2', $block_content );
```

Comment:
I wasn't sure if we wanted to remove the block attrib classnames from the `figure` element as well as adding them to the `div` wrapper. It seemed a little confusing to have them in both places, but maybe it is ok to leave them in both places?

Also, if we do decide to remove them from the `figure` then there may be a way that we can do it as part of the same `preg` that is adding the `div` - but my Friday wasn't able to work out how that regex might look 😄

Reply:
>  It seemed a little confusing to have them in both places, but maybe it is ok to leave them in both places?

I think we should remove them from the inner figure that way we match exactly the markup that was there before and avoid any surprises.

>  but if it is possible my today brain wasn't able to work out how that regex might look 

😬 I think you're asking the wrong person about regex :P I'm not an expert at all, and as long as it works, I'm ok with it. Potentially we could try to add unit tests for the `gutenberg_restore_image_outer_container` function to check different cases (I should probably have done that on the initial PR).

cc @mcsf for regex expertise :P